ConsoleReader.java

```diff
@@ -0,0 +1,29 @@
+import java.util.Scanner;
+
+public class ConsoleReader implements IReader {
+  
+  Scanner requester = new Scanner(System.in);
+
+  public String askForMultiplayer() {
+    String userInput;
+    while (true) {
+      userInput = this.requester.nextLine();
+      if (userInput.equals("c") || userInput.equals("f")) {
+        break;
+      } else {
+        System.out.println("Please enter c for computer or f for friend");
+      }
+    }
+    return userInput;
+  }
+
+  public String askForName(String name) {
+    System.out.printf("What will %s's name be?", name);
```

Comment:
You could refactor these two lines into a function: something like a
```java
private String promptForStringInput(String prompt) {
  System.out.printf(prompt);
  return(requester.nextLine());
}
```

This way, maybe at somepoint in the future you may decided to prompt for input using some other method, like via a dialog box, etc.  Then you only have to change it in this function.